File: lesson_5_2/app/src/main/java/com/kalashnyk/denys/samplemvvmapp/repository/server/ApiService.java
package com.kalashnyk.denys.samplemvvmapp.repository.server;

import com.kalashnyk.denys.samplemvvmapp.repository.database.entity.UserEntity;
import com.kalashnyk.denys.samplemvvmapp.repository.database.pojo.UserResponse;

import io.reactivex.Single;
import retrofit2.Response;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface ApiService {
    @GET("/api/android/rest/api-v2.php/records/users")
    Single<Response<UserResponse>> getUsers();

    @GET("/api/android/rest/api-v2.php/records/users/{id}")
    Single<UserEntity> getUserById(@Path("id") int id);

}

File: lesson_7/app/src/main/java/com/kalashnyk/denys/kotlinsample/repository/server/ApiService.kt
package com.kalashnyk.denys.kotlinsample.repository.server

import com.kalashnyk.denys.kotlinsample.repository.database.entity.UserEntity
import com.kalashnyk.denys.kotlinsample.repository.database.pojo.UserResponse
import io.reactivex.Single
import retrofit2.Response
import retrofit2.http.GET
import retrofit2.http.Path

interface ApiService {
    @GET("/api/android/rest/api-v2.php/records/users")
    fun getUsers(): Single<Response<UserResponse>>

    @GET("/api/android/rest/api-v2.php/records/users/{id}")
    fun getUserById(@Path("id") id: Int): Single<UserEntity>
}
